File: Orquestra.Domain/Entities/Log.cs
using Orquestra.Domain.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using static Orquestra.Utils.Fixtures.Get;

namespace Orquestra.Domain.Entities;

public sealed class Log
{
    [Key]
    public Guid LogId { get; set; }

    public required LogTypeEnum LogType { get; set; }

    public string? RequestType { get; set; } = string.Empty;

    public string? Endpoint { get; set; } = string.Empty;

    public string? Parameters { get; set; } = string.Empty;

    public string? Exception { get; set; } = string.Empty;

    public string? Description { get; set; } = string.Empty;

    public int Status { get; set; }

    public Guid? UserId { get; set; }
    [ForeignKey(nameof(UserId))]
    public User? User { get; init; }

    public DateTime CreatedDate { get; set; } = GetDate();
}

File: Orquestra.Frontend/app/api/consts/log.ts
import { BASE } from '@/app/api/fetch';
import { Guid } from 'guid-typescript';
import { iUserPaginated } from './user';

const controller = 'api/Log';

export const CONSTS_LOG = {
    get: `${BASE}/${controller}`,
    getNotification: `${BASE}/${controller}/GetNotification`
};

export interface iLog {
    logId: Guid;
    logType: string;
    requestType?: string;
    endpoint?: string;
    parameters?: string;
    exception?: string;
    description?: string;
    status: number;
    userId?: Guid;
    user?: iUserPaginated;
    createdDate: Date;
}

export interface iLogPaginated {
    output: iLog[];
    count: number;
}

export interface iLogNotificationOutput {
    logId: Guid;
    emoji: string;
    logType?: string;
    requestType?: string;
    endpointName?: string;
    rawEndpoint?: string;
    description?: string;
    story?: string;
    date: Date;
}

export interface iLogNotificationOutputPaginated {
    output: iLogNotificationOutput[];
    count: number;
}